ComponentKit/Core/CKComponent.mm

```diff
@@ -166,10 +166,11 @@ - (void)unmount
 {
   CKAssertMainThread();
   if (_mountInfo != nullptr) {
-    [_scopeHandle.controller componentWillUnmount:self];
+    CKComponentController *const controller = (CKComponentController *)_scopeHandle.controller;
```

Comment:
That's a good idea. I think scope handles having a generic interface could really help us avoid casts since most of the interop with the scope infra is through the scope handles.